db/queries.py

```diff
@@ -1,23 +1,36 @@
 import init_django_orm  # noqa: F401
+from django.db.models import QuerySet, Q, F
 
-from django.db.models import QuerySet
+from db.models import LoyaltyProgram, LoyaltyProgramParticipant, Customer
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    return LoyaltyProgram.objects.all().values_list("name", "bonus_percentage")
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.filter(
+        last_activity__gt="2021-01-01",
+        last_activity__lt="2022-01-01"
```

Comment:
use `__year`

Reply:
Updated, thank you for your advice, it's better 100%.